sdk/storage/azblob/sas/service.go

```diff
@@ -151,15 +131,117 @@ func (v BlobSignatureValues) Sign(sharedKeyCredential *SharedKeyCredential) (Que
 		signature: signature,
 	}
 
-	////User delegation SAS specific parameters
-	//if udk != nil {
-	//	p.signedOID = udk.SignedOID
-	//	p.signedTID = udk.SignedTID
-	//	p.signedStart = udk.SignedStart
-	//	p.signedExpiry = udk.SignedExpiry
-	//	p.signedService = udk.SignedService
-	//	p.signedVersion = udk.SignedVersion
-	//}
+	return p, nil
+}
+
+// SignWithUDK uses an account's UserDelegationCredential to sign this signature values to produce the proper SAS query parameters.
+func (v BlobSignatureValues) SignWithUDK(userDelegationCredential *UserDelegationCredential) (QueryParameters, error) {
```

Comment:
Thinking about this name a bit more.  I know that UDK is "user delegation key".  However, the type we use isn't a key, but a credential.  But perhaps instead of abbreviating this to UDC, what if we named it `SignWithUserDelegation`?  This is also similar to the pattern for `SignWithSharedKey`, where we use the credential's type name minus the "Credential" suffix.  What do you think?

Reply:
I agree with this! It definitely will be a lot clearer and as you mentioned, we are using a credential type here.